.github/CHANGELOG.md

```diff
@@ -62,65 +69,29 @@
 * Added a new "reversible" differentiation method which can be used in simulators, but not hardware.
   The reversible approach is similar to backpropagation, but trades off extra computation for
   enhanced memory efficiency. Where backpropagation caches the state tensors at each step during
-  a forward pass, the reversible method only caches the final pre-measurement state. Compared to 
+  a simulated evolution, the reversible method only caches the final pre-measurement state. Compared to 
   the parameter-shift method, the reversible method can be faster or slower, depending on the 
   density and location of parametrized gates in a circuit (circuits with higher density of 
   parametrized gates near the end of the circuit will see a benefit).
   [(#670)](https://github.com/XanaduAI/pennylane/pull/670)
```

Comment:
Would recommend either adding a short example of usage here, something like
```python
@qml.qnode(dev, diff_method="reversible")
def circuit(params):
    ...
```
just to show users how it is accessed from the UI (it's not very clear from the changelog entry).